i6_models/parts/conformer/feedforward.py

```diff
@@ -53,3 +58,70 @@ def forward(self, tensor: torch.Tensor) -> torch.Tensor:
         tensor = self.linear_out(tensor)  # [B,T,F]
         tensor = nn.functional.dropout(tensor, p=self.dropout, training=self.training)  # [B,T,F]
         return tensor
+
+
+@dataclass
+class ConformerPositionwiseFeedForwardV2Config(ConformerPositionwiseFeedForwardV1Config):
+    """
+    New attribute:
+        dropout_broadcast_axes: string of axes to which dropout is broadcast, e.g. "T" for broadcasting to the time axis
+                                setting to None to disable broadcasting
+    """
+
+    dropout_broadcast_axes: Optional[str] = None
+
+    def check_valid(self):
+        assert self.dropout_broadcast_axes is None or self.dropout_broadcast_axes in [
+            "B",
+            "T",
+            "BT",
+        ], "invalid value, supported are None, 'B', 'T' and 'BT'"
+
+    def __post__init__(self):
+        super().__post_init__()
+        self.check_valid()
+
+
+class ConformerPositionwiseFeedForwardV2(ConformerPositionwiseFeedForwardV1):
+    """
+    Augments ConformerPositionwiseFeedForwardV1 with dropout broadcasting
+    """
+
+    def __init__(self, cfg: ConformerPositionwiseFeedForwardV2Config):
+        super().__init__(cfg)
+
+        self.dropout = nn.Dropout1d(cfg.dropout) if cfg.dropout_broadcast_axes else nn.Dropout(cfg.dropout)
+        self.dropout_broadcast_axes = cfg.dropout_broadcast_axes
+
+    def _broadcast_dropout(self, tensor: torch.Tensor) -> torch.Tensor:
+        if self.dropout_broadcast_axes is None:
+            tensor = self.dropout(tensor)
+        elif self.dropout_broadcast_axes == "T":
+            tensor = self.dropout(tensor.transpose(1, 2)).transpose(1, 2)
+        elif self.dropout_broadcast_axes == "B":
+            tensor = self.dropout(tensor.permute(1, 2, 0)).permute(2, 0, 1)
+        elif self.dropout_broadcast_axes == "BT":
+            batch_dim_size = tensor.shape[0]
+            feature_dim_size = tensor.shape[-1]
+
+            tensor = (
+                self.dropout(tensor.reshape(-1, feature_dim_size).transpose(0, 1))
+                .transpose(0, 1)
+                .reshape(batch_dim_size, -1, feature_dim_size)
+            )
```

Comment:
As before, move the dropout logic to an own module, and then you have simply:
```suggestion
        tensor = self.dropout(tensor)
```